src/main/java/org/commcare/cases/entity/AsyncNodeEntityFactory.java

```diff
@@ -123,22 +124,29 @@ protected void prepareEntitiesInternal(List<Entity<TreeReference>> entities) {
     }
 
     @Override
-    public void cacheEntities(List<Entity<TreeReference>> entities) {
+    public void cacheEntities(List<Entity<TreeReference>> entities, Boolean skipLazyLoad) {
```

Comment:
Curiosity: is there a reason you went with Object booleans (Kotlin compatibility maybe) rather than raw values?

Reply:
Ahh It wasn't a conscious choice, also Kotlin `Boolean` does compile to primitive `boolean` as well. Therefore seems better to use primitives here - 073205487a678f944fda74223523f148990d42cc